src/Modules/ScoreboardModule/Services/ScoreboardService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using EvoSC.Common.Interfaces;
+using EvoSC.Common.Interfaces.Services;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.GameModeUiModule.Enums;
+using EvoSC.Modules.Official.GameModeUiModule.Interfaces;
+using EvoSC.Modules.Official.ScoreboardModule.Config;
+using EvoSC.Modules.Official.ScoreboardModule.Interfaces;
+
+namespace EvoSC.Modules.Official.ScoreboardModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class ScoreboardService(
+    IManialinkManager manialinks,
+    IServerClient server,
+    IScoreboardNicknamesService nicknamesService,
+    IScoreboardSettings settings,
+    IGameModeUiModuleService gameModeUiModuleService,
+    IMatchSettingsService matchSettingsService
+)
+    : IScoreboardService
+{
+    private const string ScoreboardTemplate = "ScoreboardModule.Scoreboard";
+    private const string MetaDataTemplate = "ScoreboardModule.MetaData";
+    private readonly object _currentRoundMutex = new();
+    private readonly object _isWarmUpMutex = new();
+    private int _roundNumber = 1;
+    private bool _isWarmUp;
```

Comment:
Suggestion on what you should do here:
- Move these state variables and related methods out of this class in it's own singleton service.
- Set this service as transient/scoped, depending on how its used.
- Inject the newly-created singleton into this class.